components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/IoTHubMessageSink.cs

```diff
@@ -67,11 +68,21 @@ public class IoTHubMessageSink : IMessageSink, IDisposable {
                     sw.Start();
 
                     try {
-                        if (messagesCount == 1) {
-                            await _clientAccessor.Client.SendEventAsync(messageObjects.First()).ConfigureAwait(false);
+                        if (string.IsNullOrEmpty(routingInfo)) {
+                            if (messagesCount == 1) {
+                                await _clientAccessor.Client.SendEventAsync(messageObjects.First()).ConfigureAwait(false);
+                            }
+                            else {
+                                await _clientAccessor.Client.SendEventBatchAsync(messageObjects).ConfigureAwait(false);
+                            }
                         }
                         else {
-                            await _clientAccessor.Client.SendEventBatchAsync(messageObjects).ConfigureAwait(false);
+                            if (messagesCount == 1) {
+                                await _clientAccessor.Client.SendEventAsync(routingInfo, messageObjects.First()).ConfigureAwait(false);
+                            }
+                            else {
+                                await _clientAccessor.Client.SendEventBatchAsync(routingInfo, messageObjects).ConfigureAwait(false);
+                            }
                         }
                     }
                     catch (Exception e) {
```

Comment:
Hm, don't we have a metrics counter that we should increase in exceptional cases when we fail to send the message ?

Reply:
nop...